pkg/pipelines/scm/gitlab.go

```diff
@@ -0,0 +1,95 @@
+package scm
+
+import (
+	"net/url"
+	"strings"
+
+	triggersv1 "github.com/tektoncd/triggers/pkg/apis/triggers/v1alpha1"
+)
+
+const (
+	gitlabPRBindingName   = "gitlab-pr-binding"
+	gitlabPushBindingName = "gitlab-push-binding"
+
+	gitlabCIDryRunFilters = "header.match('X-Gitlab-Event','Merge Request Hook') && body.object_attributes.state == 'opened' && body.project.path_with_namespace == '%s'  && body.project.default_branch == body.object_attributes.target_branch"
+
+	gitlabCDDeployFilters = "header.match('X-Gitlab-Event','Push Hook') && body.project.path_with_namespace == '%s' && body.ref.endsWith(body.project.default_branch)"
+)
+
+func init() {
+	supportedTriggerBindings = append(supportedTriggerBindings, gitlabPRBindingName, gitlabPushBindingName)
+}
+
+// GitLabRepository represents a service on a GitLab repo
+type GitLabRepository struct {
+	repository
+}
+
+// NewGitLabRepository returns an instance of GitLabRepository
+// NewGitLabRepository returns an instance of GitLabRepository
+func NewGitLabRepository(rawURL string) (*GitLabRepository, error) {
+	parsedURL, path, err := processRawURL(rawURL, func(parsedURL *url.URL) (string, error) {
+		components, err := splitRepositoryPath(parsedURL)
+		if err != nil {
+			return "", err
+		}
+		components[len(components)-1] = strings.TrimSuffix(components[len(components)-1], ".git")
+		path := strings.Join(components, "/")
```

Comment:
Like in the NewGitHubRepository(), we should validate the number of path components is between 2 and 5 inclusively.